### PR TITLE
feature(enhancement): GA4- allow custom dimension passthrough

### PR DESCRIPTION
**Pull request by rohithkaza**
## Description of the change

> Allowing users to send custom properties to GA4.
> To add these custom properties to the payload, we have excluded(using EXCLUSION lists for every event) some fields(which are already mapped) and then added all other fields to the event payload(using the `extractCustomFields` function) before sending it to GA4.

## Type of change
- [ ] New feature (non-breaking change that adds functionality)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by ItsSudip — DISMISSED** [review]
LGTM

File: v0/destinations/ga4/utils.js [hunk 1]
```diff
@@ -284,6 +284,29 @@ function getDestinationItemProperties(message, isItemsRequired) {
...
+      ga4ExclusionList.push(mappingSourceKeys.split(".").pop());
+    } else {
+      mappingSourceKeys.forEach(item => {
+        ga4ExclusionList.push(item.split(".").pop());
+      });
+    }
+  });
+
+  // We are mapping "products" to "items", so to remove redundancy we should not send products again
+  ga4ExclusionList.push("products");
```
Comment (akashrpo) [inline v0/destinations/ga4/utils.js:305]: We also need to add `GA4_RESERVED_PARAMETER_EXCLUSION` list here as reserved parameter names cannot be used. This list is present above.

File: v0/destinations/ga4/utils.js [hunk 2]
```diff
@@ -284,6 +284,29 @@ function getDestinationItemProperties(message, isItemsRequired) {
   return items;
 }
 
+// This function will return exclusion list for a particular event.
+// Exclusion list= sourceKeys that are already mapped
```
Comment (akashrpo) [inline v0/destinations/ga4/utils.js:288]:
```suggestion
/**
 * get exclusion list for a particular event
 * ga4ExclusionList contains the sourceKeys that are already mapped
 * @param {*} mappingJson
 * @returns
 */
```

File: v0/destinations/ga4/utils.js (hunk 1 above)
Comment (rohithkaza) [reply v0/destinations/ga4/utils.js:309]: Done

File: v0/destinations/ga4/utils.js (hunk 2 above)
Comment (rohithkaza) [reply v0/destinations/ga4/utils.js]: Done

**Review by akashrpo — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**